scripts/nbconverted/main_Pa_sim_enhance_AtoB.py

```diff
@@ -0,0 +1,225 @@
+# 
+# Hyperparmeters should include: 
+# 1. Size of gene set A
+# 2. Size of gene set B
+# 3. Proportion of gene A expression
+# 4. Thresholds
+# 5. Log file with hyperparameter selections
+#-----------------------------------------------------------------------------------------------------------------
+get_ipython().run_line_magic('load_ext', 'autoreload')
+get_ipython().run_line_magic('autoreload', '2')
+
+import os
+import pandas as pd
+import numpy as np
+import random
+import seaborn as sns
+
+from functions import generate_input, vae, def_offset, interpolate, pca, plot
+
+from numpy.random import seed
+randomState = 123
+seed(randomState)
+
+
+# In[2]:
+
+
+# Hyperparameters
+
+# Size of the gene set A
+geneSetA_size = 100
+
+# Size of the gene set that will be regulated by gene A
+geneSetB_size = 1000 
+
+# Percentage to upregulate each gene in set B
+effect_size = 0.5
+
+# Threshold for activation of gene A 
+thresholdA = 0.5
+
+# Name of analysis
+analysis_name = 'sim_AB_2775_300_skewA'
+
+
+# In[3]:
+
+
+# Load arguments
+data_file = os.path.join(os.path.dirname(os.getcwd()), "data", "all-pseudomonas-gene-normalized.zip")
+
+
+# In[4]:
+
+
+# Read in data
+data = pd.read_table(data_file, header=0, sep='\t', index_col=0, compression='zip').T
+data.head(5)
+
+
+# In[5]:
+
+
+# Randomly select genes for set A
+gene_ids = list(data.columns)
+geneSetA = random.sample(gene_ids, geneSetA_size)
+
+
+# In[6]:
+
+
+# checkpoint:  Check the number of genes
+assert(len(gene_ids)==5549)
+print("Confirmed that all gene ids are available")
+
+
+# In[7]:
+
+
+# Remove gene ids that were selected for gene set A
+gene_ids = [elem for elem in gene_ids if elem not in geneSetA]
+print("After removing {} gene ids for set A, there are {} gene ids remaining.".format(geneSetA_size, len(gene_ids)))
+
+
+# In[8]:
+
+
+# Randomly select genes for gene set B
+geneSetB = random.sample(gene_ids, geneSetB_size)
+
+# Remove gene ids that were selected for gene set B
+gene_ids = [elem for elem in gene_ids if elem not in geneSetB]
+print("After removing {} gene ids for set B, there are {} gene ids remaining.".format(geneSetB_size, len(gene_ids)))
+
+
+# In[9]:
+
+
+# checkpoint:  Check that genes in set A are distinct from genes in set B
+assert(len(set(geneSetA).intersection(set(geneSetB))) == 0)
+
+
+# In[10]:
+
+
+# Main input data simulation 
+
+# Loop through all samples
+num_samples = data.shape[1]
+for sample_id in data.index:
+    row = data.loc[sample_id]
+    
+    # Randomly select a value [0,1] where each value is uniformly likely to be chosen
+    new_A_exp = random.uniform(0.4, 1.0)
+    
+    # Set gene expression value for genes in set A to be the same random value selected
+    data.loc[sample_id][geneSetA] = new_A_exp
+    
+    # Select sample gene from set A to be representative since the expression is the same
+    # for all genes in the set
+    geneSetA_pick = geneSetA[0]
```

Comment:
Good point to improve the computational efficiency.  The location was there just because it was following my mental logic, but I have added an additional comment to hopefully make this clear.